lib/common/client/base.rb

```diff
@@ -48,15 +48,18 @@ def connection
         end.call
       end
 
-      def perform(method, path, params, headers = nil)
+      def perform(method, path, params, headers = nil, options = nil)
         raise NoMethodError, "#{method} not implemented" unless config.request_types.include?(method)
-        send(method, path, params || {}, headers || {})
+        send(method, path, params || {}, headers || {}, options || {})
       end
 
-      def request(method, path, params = {}, headers = {})
+      def request(method, path, params = {}, headers = {}, options = {}) # rubocop:disable Metrics/MethodLength
```

Comment:
think you need the enable comment at the bottom of the method as well otherwise it applies to all the subsequent methods too

Reply:
If you disable it at the end like that it only disable it for that method

Reply:
ah ok 👍

Reply:
Being yelled at by rubocop is a pretty good indication that some rework is in order. Adding rubocop exceptions is a means to an end but is 🤢

Reply:
I dont disagree... but in this case the method is only doing about 6 lines of work and the rest is taken up by request handling.

Reply:
Oh yuck, right. Definitely not a quick fix but definitely... yuck.